### DISCUSSION TITLE
Calculate speeds of cars driving by.

### DISCUSSION THREAD

**Discussion by y2kdread** [blakeblackshear/frigate#6290] [Ask A Question] [answered]
Hi - Is it possible for Frigate to calculate the speed of cars that are detected by cameras? We are having issues with people speeding on our street and it seems like this would be something that Frigate can do.

**Comment by NickM-27** [comment]
it's not directly supported but could certainly be done

**Reply by y2kdread** [reply]
Do you have any ideas on how that could be done?

**Reply by NickM-27** [reply]
Off the top of my head you could setup two zones and use the MQTT events topic https://docs.frigate.video/integrations/mqtt#frigateevents to get the time between them using frame time

**Reply by y2kdread** [reply]
Awesome, thanks! I'll have to check that out

**Reply by moskovskiy82** [reply]
Have set up two zones. They are not that apart (30 metres between them)
Any ideas which data to take?
The frame time?

```
{
  "before": {
    "id": "1689411553.371775-kvno8z",
    "camera": "roads",
    "frame_time": 1689411559.982616,
    "snapshot_time": 1689411559.782061,
    "label": "car",
    "sub_label": null,
    "top_score": 0.84375,
    "false_positive": false,
    "start_time": 1689411553.371775,
    "end_time": null,
    "score": 0.83984375,
    "box": [
      877,
      93,
      1271,
      423
    ],
    "area": 130020,
    "ratio": 1.1939393939393939,
    "region": [
      348,
      0,
      1280,
      932
    ],
    "stationary": false,
    "motionless_count": 1,
    "position_changes": 1,
    "current_zones": [
      "zone_0"
    ],
    "entered_zones": [
      "zone_1",
      "zone_0"
    ],
    "has_clip": true,
    "has_snapshot": true
  },
  "after": {
    "id": "1689411553.371775-kvno8z",
    "camera": "roads",
    "frame_time": 1689411560.38212,
    "snapshot_time": 1689411559.782061,
    "label": "car",
    "sub_label": null,
    "top_score": 0.84375,
    "false_positive": false,
    "start_time": 1689411553.371775,
    "end_time": 1689411565.581974,
    "score": 0.68359375,
    "box": [
      1084,
      152,
      1277,
      471
    ],
    "area": 61567,
    "ratio": 0.6050156739811913,
    "region": [
      364,
      0,
      1280,
      916
    ],
    "stationary": false,
    "motionless_count": 0,
    "position_changes": 1,
    "current_zones": [
      "zone_0"
    ],
    "entered_zones": [
      "zone_1",
      "zone_0"
    ],
    "has_clip": true,
    "has_snapshot": true
  },
  "type": "end"
}
```

**Reply by consolecwby** [reply]
How did this work for you. Looking for a similar solution.

**Comment by moskovskiy82** [comment]
Was going to ask the same just an hour ago.  Any other pointers?  Found an app which recognizes the front and rear wheels and does the calculation based on time it takes for them to travel and calculate based on wheelbase. Quite ingenious

**Reply by y2kdread** [reply]
what is the app that you found that does that? I've never really worked with MQTT before so something that does this out of the box would be much quicker!

**Reply by moskovskiy82** [reply]
> what is the app that you found that does that? I've never really worked with MQTT before so something that does this out of the box would be much quicker!

There is a bunch of projects on github using opencv. How to integrate them to frigate is another question.

**Reply by ghzgod** [reply]
Wouldn't your cameras FPS be a factor as well?  I am not a smart person.

**Comment by carlostipton1319** [comment]
https://homecleaningblog.com/is-it-possible-to-install-additional-safety-features-on-an-older-model-of-garage-door-opener/if you want to calculate the speeds of cars driving by, you would typically need to measure the distance traveled by the car and the time it takes to cover that distance. The formula for speed is:

Speed = Distance / Time

To calculate the speeds of cars, you would need to have a method of accurately measuring the distance traveled, such as using a measuring tape or a known reference point. You would also need a way to measure the time taken, which could be done using a stopwatch or a timing device.

By dividing the distance traveled by the time taken, you can calculate the speed of the cars driving by.

**Reply by moskovskiy82** [reply]
A chat GPT spam bot. How nice

**Comment by jvijj** [comment]
Has anyone found a simple-ish solution for this?

**Comment by ctml91** [comment]
Would be very cool to have this "Object Speed Detection" natively supported in Friday.

User draws a zone and inputs actual distance between boundaries, then frigate measures the speed and annotates the event.

**Comment by reidprichard** [comment, marked as answer]
FYI - this has been added in the [0.16.0 beta](https://github.com/blakeblackshear/frigate/releases/tag/v0.16.0-beta2).

Beta docs on the subject:
[https://deploy-preview-16390--frigate-docs.netlify.app/configuration/zones#speed-estimation](https://deploy-preview-16390--frigate-docs.netlify.app/configuration/zones#speed-estimation)
